user_guide_src/source/testing/mocking.rst

```diff
@@ -0,0 +1,53 @@
+######################
+Mocking System Classes
+######################
+
+Several classes within the framework provide mocked versions of the classes that can be used during testing. These classes
+can take the place of the normal class during test execution, often providing additional assertions to test that actions
+have taken place (or not taken place) during the execution of the test. This might be checking data gets cached correctly,
+emails were sent correctly, etc.
+
+.. contents::
+    :local:
+    :depth: 1
+
+Cache
+=====
+
+You can mock the cache with the ``mock()`` method, using the ``CacheFactory`` as its only parameter.
+::
+
+    $mock = mock(CodeIgniter\Cache\CacheFactory::class);
+
+While this returns an instance of ``CodeIgniter\Test\Mock\MockCache`` that you can use directly, it also inserts the
+mock into the Service class, so any calls within your code to ``service('cache')`` or ``Config\Services::cache()`` will
+use the mocked class within its place.
+
+When using this in more than one test method within a single file you should call either the ``clean()`` or ``bypass()``
+methods during the test ``setUp()`` to ensure a clean slate when your tests run.
+
+Additional Methods
+------------------
+
+You can instruct the mocked cache handler to never do any caching with the ``bypass()`` method. This will emulate
+using the dummy handler and ensures that your test does not rely on cached data for your tests.
```

Comment:
Reading back through the descriptions of dummies, stubs, mocks, spies, etc, I don't think we need to differentiate, personally. They can all be handled with the same object as far as I can tell. Otherwise, it's all in how you use them. Especially while we're trying to get the community to embrace testing more, I think it's better to keep the nomenclature as simple as possible.